flytekit/remote/remote.py

```diff
@@ -1068,15 +1068,19 @@ def terminate(self, execution: FlyteWorkflowExecution, cause: str):
     def _assign_inputs_and_outputs(self, execution, execution_data, interface):
         """Helper for assigning synced inputs and outputs to an execution object."""
         with self.remote_context() as ctx:
+            input_literal_map = self._get_input_literal_map(execution_data)
+            execution._raw_inputs = LiteralsResolver(input_literal_map.literals)
             execution._inputs = TypeEngine.literal_map_to_kwargs(
                 ctx=ctx,
-                lm=self._get_input_literal_map(execution_data),
+                lm=input_literal_map,
                 python_types=TypeEngine.guess_python_types(interface.inputs),
```

Comment:
This is a beta API, can we drop the outputs?

Reply:
i think I just want to support, if `as_type` is None then we can simply return the `guessed` type?
This will also prevent us from converting the type too eagerly?